### PR TITLE
CA-187943: Show a warning when a new VM's disks cannot be accommodated on the template or default SR due to lack of space.

### PR DESCRIPTION
**Pull request by kc284**
This one does not need to go into master immediately. Also, I've made 4 commits for easier review, it may be better to squash them later.

### CODE REVIEW

File: XenAdmin/Wizards/NewVMWizard/Page_Storage.cs [hunk 1]
```diff
@@ -167,88 +170,79 @@ private void LoadDisks()
...
                 return defaultSR;
+            }
 
             foreach (SR sr in connection.Cache.SRs)
             {
                 if (sr.CanCreateVmOn() &&
                     sr.CanBeSeenFrom(affinity) && sr.VdiCreationCanProceed(diskSize))
+                {
+                    if (suggestedSR != null && defaultSR != null)
+                        tooltip = Messages.NEWVMWIZARD_STORAGEPAGE_NOSPACE;
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewVMWizard/Page_Storage.cs]: When the suggestedSR is the default SR, the more suitable message would be NEWVMWIZARD_STORAGEPAGE_SUGGESTED_NOSPACE.

File: XenModel/Messages.resx [hunk 2]
```diff
@@ -9226,9 +9226,18 @@ Review these settings, then click Previous if you need to change anything. Other
...
     <value>Created by template provisioner</value>
   </data>
+  <data name="NEWVMWIZARD_STORAGEPAGE_DEFAULT_NOSPACE" xml:space="preserve">
+    <value>The default SR '{0}' does not have enough free space for the new VM's disks. [XenCenter] has selected a different SR for you.</value>
+  </data>
   <data name="NEWVMWIZARD_STORAGEPAGE_NAME" xml:space="preserve">
     <value>Storage</value>
   </data>
+  <data name="NEWVMWIZARD_STORAGEPAGE_NOSPACE" xml:space="preserve">
+    <value>Neither the default SR nor the suggested by the template SR have enough free space for the new VM's disks. [XenCenter] has selected a different SR for you.</value>
```
Comment (MihaelaStoica) [inline XenModel/Messages.resx]: I think that "the SR suggested by the template" sounds better here

File: XenAdmin/Wizards/NewVMWizard/Page_Storage.cs (hunk 1 above)
Comment (kc284) [reply XenAdmin/Wizards/NewVMWizard/Page_Storage.cs]: I realised there is one more problem in the logic, namely when the VM has no home server.

**Comment by kc284** [comment]
I have rebased the whole thing on master, please review only the last commit with the corrections.

**Pull request merged**